service/src/main/kotlin/fi/espoo/evaka/document/childdocument/ChildDocumentService.kt

```diff
@@ -34,6 +35,31 @@ class ChildDocumentNotificationService(
         asyncJobRunner.registerHandler(::sendChildDocumentNotificationEmail)
     }
 
+    fun completeAndPublishChildDocumentsAtEndOfTerm(
+        tx: Database.Transaction,
+        now: HelsinkiDateTime
+    ) {
+        val documentIds =
+            tx.createQuery<Any> {
+                    sql(
+                        """
+                SELECT cd.id
+                FROM child_document cd 
+                JOIN document_template dt on dt.id = cd.template_id
+                WHERE dt.validity << ${bind(FiniteDateRange(now.toLocalDate(), now.toLocalDate()))} AND cd.status <> 'COMPLETED'
+            """
+                            .trimIndent()
+                    )
+                }
+                .mapTo<ChildDocumentId>()
+                .list()
+
+        if (documentIds.isNotEmpty()) {
+            tx.markCompletedAndPublish(documentIds, now)
+            documentIds.forEach { scheduleEmailNotification(tx, it, now) }
```

Comment:
Is it right to send an email update about automatic closing 🤔 No "real changes" were made.

Reply:
Good question, which made me think of couple other similar cases. Will check with POs.